src/rules/noUnusedVariableRule.ts

```diff
@@ -33,18 +38,57 @@ class NoUnusedVariablesWalker extends Lint.RuleWalker {
     private skipParameterDeclaration: boolean;
     private skipVariableDeclaration: boolean;
 
+    // after a this.walk is completed, this.reactImport will be set to a NamespaceImport
+    // iff:
+    //
+    // + a react option has been provided to the rule and
+    // + an import of a module that matches one of OPTION_REACT_MODULES is found, to a
+    //   NamespaceImport named OPTION_REACT_NAMESPACE_IMPORT_NAME
+    //
+    // e.g.
+    //
+    // import * as React from "react/addons";
+    //
+    // If reactImport is defined when a walk is completed, we need to have:
+    //
+    // a) seen another usage of React and/or
+    // b) seen a JSX identifier
+    //
+    // otherwise a usage error will be reported
+    private reactImport: ts.NamespaceImport;
+    private reactUsed: boolean;
+    private seenJsx: boolean;
+
     constructor(sourceFile: ts.SourceFile, options: Lint.IOptions, languageService: ts.LanguageService) {
         super(sourceFile, options);
         this.languageService = languageService;
         this.skipVariableDeclaration = false;
         this.skipParameterDeclaration = false;
+        this.reactImport = undefined;
+        this.reactUsed = false;
+        this.seenJsx = false;
+    }
+
+    public walk(node: ts.Node): void {
+        super.walk(node);
+
+        if (this.hasOption(OPTION_REACT) && this.reactImport) {
+            if (!this.reactUsed && !this.seenJsx) {
+                let name = this.reactImport.name.getText();
+                this.addFailure(this.createFailure(this.reactImport.pos, name.length, `${Rule.FAILURE_STRING}'${name}'`));
+            }
+        }
+    }
+
+    public visitJsxElement(node: ts.JsxElement): void {
+        this.seenJsx = true;
```

Comment:
Interestingly is there a good way to enforce that `super` is called in certain situations? It not I smell another rule....